### DISCUSSION TITLE
Getting the language name from `Language` struct

### DISCUSSION THREAD

**Discussion by RRethy** [tree-sitter/tree-sitter#1955] [Q&A] [answered]
Is there any way to get the name of the language that we get back from [language_configuration_for_file_name](https://github.com/tree-sitter/tree-sitter/blob/master/cli/loader/src/lib.rs#L203)? At the moment I can only see file extensions (in the language configuration struct we get back), and I don't see anything inside `package.json` for any parser repos, is there anything I'm missing or is this not possible?

**Comment by ahlinc** [comment]
The name only exists in `grammar.js` and `grammar.json` files but doesn't stored inside of language binary file.
You can try to get `root_path` from `LanguageConfiguration` append to it `src/grammar.json` and read the name from the json file.

**Comment by amaanq** [comment, marked as answer]
You can now, starting in 0.25 (not yet released).
